### PR TITLE
Refactor of overset logic to allow multi-solver mode (2)

### PR DESCRIPTION
**Pull request by sayerhs**
Builds on #637 and does the following:

- Clean up Realm initialization to perform pre/post actions properly and let TimeIntegrator handle overset
- Remove realm specific code calls that were in TimeIntegrator and create realm pre/post work methods to handle it cleanly within Realm
- Add additional logic in overset manager to restore original code behavior for fully-coupled overset solves, #637 introduced small diffs that this pull-request restores original behavior. 

## Checklist
*All pull requests*
- [X] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [X] Linux
    - [ ] MacOS
  - Compilers 
    - [X] GCC
    - [ ] LLVM/Clang
    - [ ] Intel compilers
    - [ ] NVIDIA CUDA
- [X] Compiles without warnings
- [X] Passes all unit tests
- [X] Passes all regression tests
- [ ] Documentation builds without errors

### CODE REVIEW

File: include/overset/ExtOverset.h [hunk 1]
```diff
@@ -44,15 +44,26 @@ class ExtOverset
   void set_multi_solver_mode(const bool flag)
   { multiSolverMode_ = flag; }
 
+  bool is_external_overset() const
+  { return hasOverset_ && isExtOverset_; }
```
Comment (ashesh2512) [inline include/overset/ExtOverset.h:48]: Should `hasOverset_` be `true` if `isExtOverset_` is `true` ?

Comment (sayerhs) [reply include/overset/ExtOverset.h:48]: Thanks, that is indeed redundant. I had a more complicated logic that I replaced with `isExtOverset_` but overlooked an additional simplification.

Comment (sayerhs) [reply include/overset/ExtOverset.h]: Fixed in 81c7d132cbc4

**Comment by sayerhs** [comment]
@ashesh2512 Since @alanw0 hasn't reviewed it yet... I added one more commit. Can you please take a look at that: 
5eeb1c6

**Pull request merged**